### DISCUSSION TITLE
Truncate inlay hints to custimizable max length

### DISCUSSION THREAD

**Discussion by meliache** [joaotavora/eglot#1176] [Ideas]
I used Eglot for LaTeX with the TeXLab language server and was at first happy to find out it supports inlay hints, but then saw that they are sometimes overly long, e.g. printing the entire figure caption in the inlay hint to a figure label (hint in grey):
![](https://user-images.githubusercontent.com/5121824/222107451-0f087d6b-1ca1-4a04-afc5-d6f571873c83.png)
This definetly doesn't enhance readability. I assume this is probably not an Eglot but, but an issue with the language server, which doesn't have customization options for the inlay hints. I opened an [TeXLab issue](https://github.com/latex-lsp/texlab/issues/858) for that.
 
But what Eglot might do, is to offer a way to truncate inlay hints, e.g. in a similar way to how Eglot truncates messages. Ideally a well-behaved server should provide concise hints where truncation not necessary, but I assume the Eglot implementation how to deal with that should be rather simple, which is why I suggest this feature.

I'm wondering why no other TeXLab user had this issue, maybe they are using LSP clients which either don't support inlay hints or other clients already have a way to truncate or hide long inlay hints. I just created the TeXLab issue and will see what their feedback will be.

**Comment by meliache** [comment]
The issue of the overly long inlay hints in TeXLab due to the caption being included was fixed in https://github.com/latex-lsp/texlab/commit/dd85dce46f20bf944616c7929cc7b6fe81c1ff80. Personally, for me, an eglot feature to truncate inlay hints are is not necessary anymore.

However, I would leave this discussion open whether we would like to have the option of client-side inlay hint truncation anyway in case some other servers show overly long messages.

**Reply by joaotavora** [reply]
Such an option, at least as a user-visible thing, would be a bit excessive, IMO.
